backend/tools/websocket/websocket_client.py

```
@@ -0,0 +1,148 @@
+                "Server rejected connection: HTTP %s. Check URI, headers, and server logs.",
+                e.status_code,
+            )
+            if e.status_code == 403:
+                logger.error(
+                    "HTTP 403 Forbidden - Likely requires Origin header (removed for testing TypeError)."
+                )
+            break  # Stop retrying on definitive rejection like 403
+        except websockets.exceptions.InvalidURI:
+            logger.error("Invalid WebSocket URI: %s", uri)
+            break  # Stop if URI is fundamentally wrong
+        except (ConnectionRefusedError, websockets.exceptions.InvalidStatusCode) as e:
```

Ruff (0.11.9)

try-except block with duplicate exception `websockets.exceptions.InvalidStatusCode`

(B025)